siliconcompiler/core.py

```diff
@@ -787,6 +787,15 @@ def get(self, *keypath, field='value', job=None, cfg=None):
 
         """
 
+        # Ensure that all keypath values are strings.
+        # Scripts may accidentally pass in [None] if a prior schema entry was unexpectedly empty.
+        for kp in keypath:
+            if not type(kp) == str:
+                self.error(\
+f'''An invalid keypath was passed to "chip.get":
+{keypath}
+Your Chip configuration may be missing a parameter which is expected by your build script.''')
```

Comment:
How about formatting like?
```python
self.error(
   'An invalid keypath was passed to chip.get: \n'
   f'{keypath} '
   ...
)
```
I think the full de-indent is a bit jarring stylistically.

I'd also be wary of multi-line error messages, since they kind of break the table formatting when logged. Maybe we could handle this by allowing self.error() to accept a list of messages and then log each one individually?

Reply:
I also wonder if it's a better idea to make the error message hew more closely to the error and be less speculative? Something like a tweaked version of the exisisting error message in `chip.add()` seems useful enough to me:

```
"Key [{key}] in keypath  {keypath} is not a string" 
```

I imagine seeing the invalid keypath and the call location in the traceback is probably enough to infer, although it does make sense to strive to be as helpful as possible with error messages.

Reply:
Good point, I re-formatted the error message a bit.